accord-core/src/main/java/accord/local/Node.java

```diff
@@ -241,7 +242,21 @@ public long epoch()
     private synchronized EpochReady onTopologyUpdateInternal(Topology topology, boolean startSync)
     {
         Supplier<EpochReady> bootstrap = commandStores.updateTopology(this, topology, startSync);
-        return this.topology.onTopologyUpdate(topology, bootstrap);
+        Supplier<EpochReady> ordering = () -> {
+            if (this.topology.isEmpty()) return bootstrap.get();
+            return order(this.topology.epochReady(topology.epoch() - 1), bootstrap.get());
+        };
+        return this.topology.onTopologyUpdate(topology, ordering);
+    }
+
+    private static EpochReady order(EpochReady previous, EpochReady next)
+    {
+        if (previous.coordination.isDone()) return next;
```

Comment:
if we did `flatMap` 100% of the time then everything is linked together, so this allows the more common case (epochs don't all happen at once) to avoid linking